pkg/templateservicebroker/api/types.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// +genclient=true
+
+type TemplateProvision struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec   TemplateProvisionSpec
+	Status TemplateProvisionStatus
+}
+
+type TemplateProvisionSpec struct {
```

Comment:
needs an impersonation id field so the controller knows who to impersonate when instantiating the template?